packages/lib/models/Setting.ts

```diff
@@ -1222,6 +1222,18 @@ class Setting extends BaseModel {
 				isGlobal: true,
 			},
 
+			hideMenuBar: {
+				value: false,
+				type: SettingItemType.Bool,
+				section: 'application',
+				public: true,
```

Comment:
It might make sense to hide this setting on MacOS (as [the documentation marks that method as Windows and Linux only](https://www.electronjs.org/docs/latest/api/browser-window#winsetautohidemenubarhide-windows-linux)).

To see how this is done, we do something similar for the `autoUpdate` setting on Linux:
https://github.com/laurent22/joplin/blob/f8a25cb275a0beb9ba3ef9bf4271d38947449f42/packages/lib/models/Setting.ts#L1384

(The `autoUpdate` setting probably shouldn't be hidden on Linux, so the above example may cause a bug.)

Reply:
Good point, thanks! 

I've added a condition similar to the one on `autoUpdateEnabled`, as you've mentioned.